frontend/pages/books/[id].js

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { useRouter } from "next/router";
+import {
+  Breadcrumbs,
+  EmptyState,
+  Flex,
+  Layout,
+  Section,
+  Text,
+} from "../../components";
+import { fontWeight } from "../../theme";
+import { useGetBook } from "../../api/books";
+
+const BooksDetailPage = () => {
+  const router = useRouter();
+  const { book, bookLoading, bookError } = useGetBook(router.query.id);
+  return (
+    <Layout>
+      {bookLoading || bookError || !Object.keys(book).length ? (
+        <Section>
+          <EmptyState loading={bookLoading} error={bookError} />
```

Comment:
Can you elaborate on the approach used here?

Reply:
The books detail page will render empty state if the book data is being loaded from the API, or there was an error, or no book data were loaded. I used conditional rendering so that it would prevent the part of the page that depends on the book data from being rendered in any of the cases mentioned above.